src/theory/arith/arith_msum.cpp

```diff
@@ -0,0 +1,371 @@
+/*********************                                                        */
+/*! \file arith_msum.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of arith_msum
+ **/
+
+#include "theory/arith/arith_msum.h"
+
+#include "theory/rewriter.h"
+
+using namespace CVC4::kind;
+
+namespace CVC4 {
+namespace theory {
+
+bool ArithMSum::getMonomial(Node n, Node& c, Node& v)
+{
+  if (n.getKind() == MULT && n.getNumChildren() == 2 && n[0].isConst())
+  {
+    c = n[0];
+    v = n[1];
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+bool ArithMSum::getMonomial(Node n, std::map<Node, Node>& msum)
+{
+  if (n.isConst())
+  {
+    if (msum.find(Node::null()) == msum.end())
```

Comment:
Is it possible to merge these two if conditions?

Reply:
No, the top level if-then-else is splitting on the cases for n and the inner ifs are checking whether we can constructor an msum from n.